app/models/spree/variant_decorator.rb

```diff
@@ -86,7 +86,13 @@ def self.indexed
   end
 
   def self.active(currency = nil)
-    where(id: joins(:prices).where(deleted_at: nil).where('spree_prices.currency' => currency || Spree::Config[:currency]).where('spree_prices.amount IS NOT NULL').select("spree_variants.id"))
+    # "where(id:" is necessary so that the returned relation has no includes
```

Comment:
could you elaborate a bit more? which includes and what does the `id IN ()` do? if we could get rid of it it could be a perf. gain. which is pretty needed. I don't mind investigating this further in another issue though.

Reply:
the includes are prices, the IN clause makes the method return variants only without prices which makes the relation NOT be readonly.
I dont think there's a significant performance impact.